### PR TITLE
Async video engine 3

### PR DESCRIPTION
**Pull request by MartinDelille**
Hi @tlecomte !

I merge my last commit into your work. I need to go furthermore to check it all but I already have a few issue where I'd like your feedback. Hope I can merge this soon :-)

### CODE REVIEW

File: libs/PhVideo/PhVideoEngine.cpp [hunk 1]
```diff
@@ -9,42 +9,50 @@
...
+	PhVideoDecoder *decoder = new PhVideoDecoder();
+	decoder->moveToThread(&_decoderThread);
+	connect(&_decoderThread, &QThread::finished, decoder, &QObject::deleteLater);
+	connect(this, &PhVideoEngine::decodeFrame, decoder, &PhVideoDecoder::decodeFrame);
+	connect(this, &PhVideoEngine::openInDecoder, decoder, &PhVideoDecoder::open);
+	connect(this, &PhVideoEngine::closeInDecoder, decoder, &PhVideoDecoder::close);
+	connect(decoder, &PhVideoDecoder::frameAvailable, this, &PhVideoEngine::frameAvailable);
+	connect(decoder, &PhVideoDecoder::opened, this, &PhVideoEngine::decoderOpened);
+	connect(decoder, &PhVideoDecoder::openFailed, this, &PhVideoEngine::openInDecoderFailed);
+	_decoderThread.start();
```
Comment (MartinDelille) [inline libs/PhVideo/PhVideoEngine.cpp]: @tlecomte we have a memory leaks here. I'm thinking about having *_decoder* as a class member, what's your thought about it?

**Comment by tlecomte** [comment]
@MartinDelille Hey, it's good to see you looking at this! May I suggest to work from #319 instead? It's really much better than #318, from both features and design points of view.  Sorry,  I should probably have closed #318...

**Comment by MartinDelille** [comment]
Ok I'll look at it soon!

**Comment by MartinDelille** [comment]
I checked and I worked from the tlecomte:tlAsyncVideoEngine2 branch from #319

File: libs/PhVideo/PhVideoEngine.cpp (hunk 1 above)
Comment (MartinDelille) [inline libs/PhVideo/PhVideoEngine.cpp]: @tlecomte as I told you, this code is from #319. Do you have any thought about this memory lead? As I told you above, I'm thinking about putting *_decoder* as a member variable.

Comment (tlecomte) [inline libs/PhVideo/PhVideoEngine.cpp]: @MartinDelille Sure moving to a member variable is ok.

**Comment by tlecomte** [comment]
Well, there's something strange here. The latest commit from me in this branch is "Fix how we know if the frame has to be requested or not", but if you look at #319 you'll see that there are about 15 more commits after that.... That's why I thought at first that this was #318. Anyway there is something wrong with this selection of commits!

**Comment by MartinDelille** [comment]
Yes it's my fault: I didn't fetch your last commits when merging into my branch. I'll fix it!

**Comment by MartinDelille** [comment]
replace by #345

File: libs/PhVideo/PhVideoEngine.cpp (hunk 1 above)
Comment (MartinDelille) [inline libs/PhVideo/PhVideoEngine.cpp]: in fact there is no memory leaks since you are triggering the *deleteLater()* slot when the thread is finished.

**Pull request closed without merging**